### PR TITLE
[LoDash/fp] [Typescript 3.1+] Add LodashConvertible interface to allow calling .convert({}) on each lodash/fp function

### LINKED ISSUE #27194: Lodash/fp convert method not typed
The `convert` function that should be available on each `lodash/fp` exported member is not typed correctly for `mapValues` (the only one I checked but probably others too)

I should be able to do the following:
```js
import { mapValues } from 'lodash/fp'

const mapValuesUncapped = mapValues.convert({ cap: false })
// I get the error => `[ts] Property 'convert' does not exist on type 'LodashMapValues'.`
```

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @bczengel @chrootsu @stepancar @aj-r @Ailrun @e-cloud @thorn0 @jtmthf @DomiR

### PR DESCRIPTION
**Pull request by toniopelo**
Fixes: #27194

Please fill in this template.
Here a link to the [lodash/fp guide](https://github.com/lodash/lodash/wiki/FP-Guide#convert) talking about the `.convert({})` method I just typed in this PR.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

----

This is the same PR as #44877 but for typescript 3.1+ that was in a separate folder, I didn't catch it at the time.

### CODE REVIEW

File: types/lodash/ts3.1/scripts/generate-fp.ts [hunk 1]
```diff
@@ -848,7 +856,13 @@ function getInterfaceName(baseName: string, overloadId: number, index: number, t
     return interfaceName;
 }
 
-function interfaceToString(interfaceDef: Interface): string {
+function interfaceToString(interfaceDef: Interface, exportedInterface: boolean): string {
+// Exported interface extends LodashConvertible to allow
```
Comment (aj-r) [inline types/lodash/ts3.1/scripts/generate-fp.ts:860]: Fix indentation

**Review by aj-r — APPROVED** [review]
Looks good

**Comment by toniopelo** [comment]
@aj-r Updated :)

**Comment by amcasey** [comment]
The CI failure doesn't make sense to me.  I'm following up.

**Comment by amcasey** [comment]
I take it back - the CI errors look correct.  Please have a look.

**Comment by aj-r** [comment]
Strange, it seems like there are a lot of changes to `fp.d.ts` that shouldn't be related to this CL. It's probably a bug in the script, but I don't have time to investigate how to fix it right now. Are there any significant differences between the <3.1 script and the >=3.1 script?

**Comment by toniopelo** [comment]
Yeah a lot of changes in `fp.d.ts` seems unrelated to my changes in `generate-fp.ts`.
Is it possible that a previous PR was merged without the up-to-date `ts3.1/fp.d.ts` committed ?

**Comment by aj-r** [comment]
Yes that's possible

**Comment by TkDodo** [comment]
> > Is it possible that a previous PR was merged without the up-to-date ts3.1/fp.d.ts committed ?

> Yes that's possible



what can we do to fix the CI failures?

**Pull request closed without merging**